src/coreclr/gc/unix/gcenv.unix.cpp

```diff
@@ -1248,20 +1271,23 @@ uint64_t GetAvailablePhysicalMemory()
 
     // Get the physical memory available.
 #if defined(__APPLE__)
-    vm_size_t page_size;
-    mach_port_t mach_port;
-    mach_msg_type_number_t count;
-    vm_statistics_data_t vm_stats;
-    mach_port = mach_host_self();
-    count = sizeof(vm_stats) / sizeof(natural_t);
-    if (KERN_SUCCESS == host_page_size(mach_port, &page_size))
+    uint32_t mem_free = 0;
+    size_t mem_free_length = sizeof(uint32_t);
+    if (g_kern_memorystatus_level_mib != NULL)
```

Comment:
if `GCToOSInterface::Initialize` succeeded, this should always be non null, correct? if so we can just assert.